src/Internal.hh

```diff
@@ -53,6 +53,37 @@
+		if (buf == nullptr) {                                                                                          \
+			ZKC_LOG_WARN_NULL(#cls "_save");                                                                           \
+			return;                                                                                                    \
+		}                                                                                                              \
+                                                                                                                       \
+		try {                                                                                                          \
+          	slf->save(buf, static_cast<zenkit::ArchiveFormat>(fmt));                                                   \
+		} catch (std::exception const& exc) {                                                                          \
+			ZKC_LOG_ERROR(#cls "_save() failed: %s", exc.what());                                                      \
+			return;                                                                                                    \
+		}                                                                                                              \
+	}
+
+#define ZKC_PATH_SAVER(cls)                                                                                            \
+	void cls##_savePath(ZkCutsceneLibrary* slf, ZkString path, ZkArchiveFormat fmt) {                                  \
+		if (path == nullptr) {                                                                                         \
+			ZKC_LOG_WARN_NULL(#cls "_savePath");                                                                       \
+			return;                                                                                                    \
+		}                                                                                                              \
+                                                                                                                       \
+		try {                                                                                                          \
+            auto buf = zenkit::Write::to(path);                                                                        \
+            slf->save(buf.get(), static_cast<zenkit::ArchiveFormat>(fmt));                                             \
+		} catch (std::exception const& exc) {                                                                          \
+			ZKC_LOG_ERROR(#cls "_savePath() failed: %s", exc.what());                                                  \
+			return;                                                                                                    \
+		}                                                                                                              \
+	}
+
```

Comment:
You need to `#include "zenkit-capi/Archive.h"` so that `ZkArchiveFormat` becomes available in all files using the `_SAVER` macros.

Reply:
Done